source/extensions/filters/network/http_connection_manager/dependency_manager.cc

```diff
@@ -0,0 +1,31 @@
+#include "extensions/filters/network/http_connection_manager/dependency_manager.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace HttpConnectionManager {
+
+using envoy::extensions::filters::common::dependency::v3::Dependency;
+
+bool DependencyManager::decodePathIsValid() {
+  auto cmp = [](Dependency a, Dependency b) { return a.name() != b.name(); };
+  std::set<Dependency, decltype(cmp)> satisfied(cmp);
```

Comment:
Why not just make this a `absl::flat_hash_set<std::string>`?

Reply:
```/opt/llvm/bin/../include/c++/v1/__functional_base:54:21: error: invalid operands to binary expression ('const envoy::extensions::filters::common::dependency::v3::Dependency' and 'const envoy::extensions::filters::common::dependency::v3::Dependency')
```

Proto comparison doesn't work properly. In the future I'll probably switch this over to use `message_differencer.h`.

Reply:
For both this and the above point, I don't think we need to store the full dependency in the set, but just its name?

Reply:
I think I see what the issue is. You ideally are showing that the dependency matches the provider. You need to compare _both_ name and also the `DependencyType`. However in this PR so far you only have name. 

I suggest following the pattern of `LocalityWeightsMap` in `eds.h` for a pattern here around custom comparators, Absl, and proto keys.

Reply:
I looked at `LocalityWeightsMap` and talked to Ashley about it, and I think got a simpler way to do this. Since it seems an intermediary type is unavoidable, I went with this:
```
using DependencyTuple = std::tuple<const std::string&, int>;
absl::flat_hash_set<DependencyTuple> satisfied;
```

This isn't resilient to modifications to the Dependency proto but I don't really anticipate more fields being added so I think this works for now.